applications/nrf_desktop/configuration/common/hid_report_desc.h

```diff
@@ -84,6 +84,34 @@ static const uint8_t output_reports[] = {
 	REPORT_ID_KEYBOARD_LEDS,
 };
 
+/* Internal definitions used to calculate size of the biggest supported HID input report. */
+#define _REPORT_BUFFER_SIZE_MOUSE \
+	(IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_MOUSE_SUPPORT) ? (REPORT_SIZE_MOUSE) : (0))
+#define _REPORT_BUFFER_SIZE_KEYBOARD_KEYS \
+	(IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_KEYBOARD_SUPPORT) ? (REPORT_SIZE_KEYBOARD_KEYS) : (0))
+#define _REPORT_BUFFER_SIZE_SYSTEM_CTRL								\
+	(IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_SYSTEM_CTRL_SUPPORT) ?				\
+	 (REPORT_SIZE_SYSTEM_CTRL) : (0))
+#define _REPORT_BUFFER_SIZE_CONSUMER_CTRL							\
+	(IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_CONSUMER_CTRL_SUPPORT) ?				\
+	 (REPORT_SIZE_CONSUMER_CTRL) : (0))
+#define _REPORT_BUFFER_SIZE_BOOT_MOUSE \
+	(IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_MOUSE) ? (REPORT_SIZE_MOUSE_BOOT) : (0))
+#define _REPORT_BUFFER_SIZE_BOOT_KEYBOARD \
+	(IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_KEYBOARD) ? (REPORT_SIZE_KEYBOARD_BOOT) : (0))
+
+/* Size of the biggest supported HID input report that is part of input reports map. */
+#define REPORT_BUFFER_SIZE_INPUT_REPORT								\
+	MAX(MAX(_REPORT_BUFFER_SIZE_MOUSE, _REPORT_BUFFER_SIZE_KEYBOARD_KEYS),			\
```

Comment:
consider creating MAX6 macro:
#define MAX6(a, b, c, d, e, f) (MAX(MAX(MAX((a), (b)), MAX((c), (d))), MAX((e), (f))))

and using it for greater clarity

Reply:
The macro would be nice to have, but I think it should be part of Zephyr (not nRF Desktop application). We would also need to define `MAX_` variants for other number of arguments (or even make it `VA_ARGS` macro). Implementing it in generic way may take some work and I am not sure if it's worth to work on it on our end.

Reply:
We can add just this one here and remove it once there is such macro in zephyr

Reply:
The header content is visible for application code. I am a bit against adding only `MAX6` here because of that (macro name would suggest that it's something more generic)

Reply:
On the other hand, providing `MAX6` suggests that other counts could be supported in a similar way. @kapi-no, @mkapala-nordic, @alstrzebonski, what do you think ?

Reply:
don't know why having MAX6 suggest that others macros are supported

Reply:
If it is defined at the file level then why would it suggest that others are supported?
Personally it is fine for me either way, but prefer leaving it as is.

Reply:
I like the MAX6 idea. It would be defined locally just in this file as a helper.

Reply:
Keep in mind that C sources including the header would "see" the definition too. Also, the comment explicitly mentions how the macro works, so it should be quite understandable

Reply:
> I like the MAX6 idea. It would be defined locally just in this file as a helper.

I tried to define `MAX_VA_ARGS` macro that would allow to select maximum from any number of arguments, but it seems tricky to do it in generic way (solution still does not work for me). I prefer not to define only `MAX6` macro here and stick with the existing approach if possible then.